model/oauth/client.go

```diff
@@ -426,6 +445,20 @@ func (c *Client) Update(i *instance.Instance, old *Client) *ClientRegistrationEr
 		c.NotificationDeviceToken = old.NotificationDeviceToken
 	}
 
+	// Updating metadata
+	appSlug := c.SoftwareID
+	if strings.HasPrefix(c.SoftwareID, "registry://") {
+		appSlug = strings.TrimPrefix(c.SoftwareID, "registry://")
+	}
+
+	md, _ := metadata.NewWithApp(appSlug, c.SoftwareVersion, DocTypeVersion)
```

Comment:
[nit] You can move this line inside the `if` as it is neither used in the `else` branch or after the if/else block